elements/lisk-validator/src/errors.ts

```diff
@@ -31,11 +30,85 @@ export interface ErrorObject {
 	data?: never;
 }
 
+interface ErrorParams {
+	[key: string]: unknown;
+}
+
+const errorParamToString = (
+	param: string | Buffer | BigInt | undefined | unknown,
+): string => {
+	let paramAsString = '';
+	if (typeof param === 'bigint') {
+		paramAsString = param.toString();
+	} else if (Buffer.isBuffer(param)) {
+		paramAsString = param.toString('base64');
+	} else if (param === undefined) {
+		paramAsString = '';
+	} else {
+		paramAsString = param as string;
+	}
+	return paramAsString;
+};
+
+const errorFormatter = (error: ErrorObject): string => {
+	let errorMessage = '';
+	switch (error.keyword) {
```

Comment:
All looks good, except I don't feel comfortable with `switch` here, it will be called here a lot of times and conditional control flow are not good if we can achieve the same with a hash map. 

If you used it because to the `default` case, that we can also achieve via. 

```js
const defaultErrorFormater = error => (error.message ?? 'unspecified validator error');
const errorFormatMaps = {
   type: error => `Property '${
 				error.dataPath ?? ''
 			}' should be of type '${errorParamToString(error.params.type)}'`,

   additionalProperties: error => `Property '${
 				error.dataPath ?? ''
 			}' has extraneous property '${errorParamToString(
 				error.params.additionalProperty,
 			)}'`
}

const errorFormatter = (error: ErrorObject): string => (errorFormatMaps[error.keyword] | defaultErrorFormater)(error);
```